src/resilience/exec_space/openMP/OpenMPResParallel.hpp

```diff
@@ -87,38 +107,84 @@ class ParallelFor< FunctorType
                  , Kokkos::RangePolicy< Traits... >
                  , KokkosResilience::ResOpenMP>{
  private:
-  using Policy    = Kokkos::RangePolicy<Traits...>;
-  using WorkTag   = typename Policy::work_tag;
+  using Policy       = Kokkos::RangePolicy<Traits...>;
+  using WorkTag      = typename Policy::work_tag;
+  using LaunchBounds = typename Policy::launch_bounds;
+  using Member       = typename Policy::member_type;
 
   const FunctorType &  m_functor;
   const Policy m_policy;
 
   ParallelFor() = delete ;
   ParallelFor & operator = ( const ParallelFor & ) = delete ;
 
-  using surrogate_policy = Kokkos::RangePolicy < Kokkos::OpenMP, WorkTag >;
+  using surrogate_policy = Kokkos::RangePolicy < Kokkos::OpenMP, WorkTag, LaunchBounds>;
+
+#ifdef KR_ENABLE_WRAPPER
+  auto MakeWrapper (int64_t work_size, int64_t offset, const FunctorType &m_functor_0, const FunctorType &m_functor_1) const{
+    if constexpr (std::is_void_v<WorkTag>){
+      auto wrapper_functor = [&, work_size, offset](int64_t i){
+        if (i < work_size)
+        {
+          m_functor (i + offset);
+        }
+        else if (( work_size <= i) && (i < (2 * work_size)))
+        {
+          m_functor_0 (i + offset - work_size);
+        }
+        else
+        {
+          m_functor_1 (i + offset - ( 2 * work_size));
+        }
+      };
+      return wrapper_functor;
+    }else if constexpr (!std::is_void_v<WorkTag>)
```

Comment:
The formatting is a bit inconsistent here, can you run clang-format on the files? (We have a .clang-format file)

Reply:
We decided to just manually do some of the formatting fixes as clang-format might generate a bunch of changes. We'll do that separately on the codebase in a different PR